loading.bs

```diff
@@ -0,0 +1,992 @@
+
+The result of <dfn>creating a header list from the CBOR map</dfn> |map| is
+returned by the following steps:
+
+1. Let |headers| be a new empty [=header list=].
+1. For each |key| → |value| of |map|:
+    1. If |key| starts with `` `:` ``, continue.
+    1. If the [=isomorphic decoding=] of |key| contains any [=ASCII upper
+        alpha=], return a failure.
+    1. If |key| doesn't match the constraints on a [=header/name=] or |value|
+        doesn't match the constraints on a [=header/value=], return a failure.
+    1. Assert: |headers| [=header list/does not contain=] |key|.
+    1. [=header list/Append=] |key|/|value| to |headers|.
+1. Return |headers|.
+
+<h3 algorithm id="read-a-body">Creating the response stream.</h3>
+
+To <dfn lt="reading a body|read a body">read a body</dfn> from a [=read buffer=]
+|stream| into a [=response=] |response| using an [=exchange signature=]
+|signature| to check its integrity, the UA MUST:
+
+1. If |signature|'s [=exchange signature/integrity header=]'s first item is:
+
+    <dl class="switch">
+
+    : "mi-draft2"
+    ::
+        1. If the length of |signature|'s [=exchange signature/integrity header=]
+            isn't 1, return a failure.
+        1. If |response|'s [=response/header list=] [=header list/does not
+            contain=] `` `mi-draft2` ``, return a failure.
+        1. Let |mi| be the [=header/combined value=] of `` `mi-draft2` `` in
+            |response|'s [=response/header list=].
+        1. If |mi| doesn't have an [=mi-sha256 parameter=], return a failure.
+        1. If [=extracting header list values=] for `` `content-encoding` `` in
+            |response|'s [=response/header list=] doesn't include
+            `` `mi-sha256` ``, return a failure.
+        1. Assert: [=Handle content codings=] used the value of the
+            [=mi-sha256 parameter=] as the [=integrity proof for the first
+            record=] when decoding the `mi-sha256` content encoding to produce
+            the bytes in |stream|.
+
+    : Anything else
+    :: Return a failure.
+
+    </dl>
+1. Let |body| be a new [=body=].
+1. Let |cancel| be the following steps, taking |reason| as an argument:
+    1. [=ReadableStream/Cancel=] |stream|'s [=read buffer/stream=] with
+        |reason|.
+1. Let |outputStream| be the result of [=ReadableStream/construct a
+    ReadableStream object|constructing a ReadableStream=] with |cancel|.
+1. Set |body|'s [=body/stream=] to |outputStream|.
+1. Set |response|'s [=response/body=] to |body|.
+1. [=In parallel=]:
+    1. [=read buffer/Dump=] |stream| to |outputStream|.
+
+
+<h3 algorithm id="request-matching">Request matching</h3>
+
+A [=request=] |browserRequest| <dfn>matches the stored request</dfn>
+|storedRequest| if the following steps return "match":
+
+1. If |browserRequest|'s [=request/url=] is not [=url/equal=] to
+    |storedRequest|'s [=request/url=], return "mismatch".
+
+1. Issue: Flesh this out, along the lines of
```

Comment:
Note that the way this is rendered in Safari looks a bit weird, I only see step 1. and 3. with a big ISSUE section in between. But I guess once Step 2 has been fleshed out, it will look ok.